client/query.go

```diff
@@ -70,14 +46,112 @@ func (c *QueryClient) Validate() error {
 }
 
 // QueryEvents initiates an Insights query, returns a response for parsing
-func (c *QueryClient) QueryEvents(nrqlQuery string) (*QueryResponse, error) {
+func (c *QueryClient) QueryEvents(nrqlQuery string) (response *QueryResponse, err error) {
+	response = &QueryResponse{}
+	err = c.Query(nrqlQuery, response)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// Query initiates an Insights query, with the JSON parsed into 'response'
+func (c *QueryClient) Query(nrqlQuery string, response interface{}) (err error) {
```

Comment:
```suggestion
func (c *QueryClient) Query(nrqlQuery string (res response, err error) {

	err = c.generateQueryURL(nrqlQuery)
	if err != nil {
		return res, err
	}

	c.Logger.Debugf("Querying: %s", nrqlQuery)
	c.generateQueryURL(nrqlQuery)
	err = c.queryRequest(response)
	if err != nil {
		return err
	}

	return c.queryRequest(nrqlQuery)

}
```

Reply:
The point of this method is to accept a result struct to populate, moving response to a return value would not work.